### PR TITLE
Add string utilities to create random capitalization of input strings

### PR DESCRIPTION
**Pull request by sleberknight**
Add several new methods in KiwiStrings2 that can generate unique sets containing differently-capitalized variants of a given input string. Add a maximum number of unique strings that can be generated which is 10,000 by default but can be changed (at the user's peril).

### CODE REVIEW

File: src/main/java/org/kiwiproject/beta/base/KiwiStrings2.java [hunk 1]
```diff
@@ -83,4 +106,193 @@ public static String replaceNullCharacters(@Nullable String str, String replacem
...
+     *
+     * @param input  the input string to randomly capitalize
+     * @param locale the Locale to use to conversion to upper and lower case
+     * @return a string marching the input string but with random capitalization
+     */
+    public static String randomlyCapitalize(@NonNull String input, @NonNull Locale locale) {
+        checkArgumentNotBlank(input);
+        checkArgumentNotNull(locale);
+
+        var rand = ThreadLocalRandom.current();
```
[the remark this answers is not in the document]
Comment (sleberknight) [reply src/main/java/org/kiwiproject/beta/base/KiwiStrings2.java:248]: @sonatype-lift ignore

[the remark this answers is not in the document]
Comment (sleberknight) [reply src/main/java/org/kiwiproject/beta/base/KiwiStrings2.java:248]: This is not a security-relevant context, so using `ThreadLocalRandom` is fine.

File: src/main/java/org/kiwiproject/beta/base/KiwiStrings2.java [hunk 2]
```diff
@@ -83,4 +106,193 @@ public static String replaceNullCharacters(@Nullable String str, String replacem
...
+        return randomCaseVariants(input, 3, locale);
+    }
+
+    /**
+     * Get the current maximum number of random strings that can be generated using
+     * {@link #randomCaseVariants(String, int)} and {@link #randomCaseVariants(String, int, Locale)}.
+     *
+     * @return the maximum number of random strings that can be generated
+     */
+    @Synchronized
```
Comment (dsingley) [inline src/main/java/org/kiwiproject/beta/base/KiwiStrings2.java:139]: why does reading the value have to be synchronized?

**Review by dsingley — CHANGES_REQUESTED** [review]

File: src/main/java/org/kiwiproject/beta/base/KiwiStrings2.java (hunk 2 above)
Comment (sleberknight) [reply src/main/java/org/kiwiproject/beta/base/KiwiStrings2.java:139]: What if some other thread is writing a new value while your thread is reading it?

Comment (dsingley) [reply src/main/java/org/kiwiproject/beta/base/KiwiStrings2.java:139]: you might get an outdated value

**Review by dsingley — APPROVED** [review]

**Pull request merged**